### PR TITLE
Folders integration

### PR DESCRIPTION
**Pull request by imod**
@cyrille-leclerc I worked on the folders integrtation - would be very happy if you could have a look at this and maybe do some test with it too.

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
I'm getting the following exception when trying to use a config file declared in a parent folder

See details in https://gist.github.com/cyrille-leclerc/5571fdc443d6f7bff4e5ec10d614a15d

I'm investigating. I'll try to get help from @jglick

**Comment by imod** [comment]
@cyrille-leclerc many thanks! 
about the FB issues... I'm still trying to figure out how to run these tests locally. This did not really help me: https://wiki.jenkins-ci.org/display/JENKINS/FindBugs+in+plugins

**Comment by cyrille-leclerc** [comment]
@domi I run the following commands to fix findbugs issues:

```
mvn findbugs:findbugs findbugs:gui
```

**Comment by imod** [comment]
@cyrille-leclerc oh wow - thats very nice! ...I'll add that to the documentation, thanks!

**Comment by cyrille-leclerc** [comment]
@imod you're welcome.

**Comment by cyrille-leclerc** [comment]
I proposed #20 to make findbugs much more happy

**Comment by cyrille-leclerc** [comment]
@imod I have created #21 and I have asked @jglick and @stephenc to review this change.

**Comment by imod** [comment]
@cyrille-leclerc FYI: I added a test case that tests your scenario

File: pom.xml [hunk 1]
```diff
@@ -55,6 +57,11 @@
     <dependencies>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>cloudbees-folder</artifactId>
+            <version>5.12</version>
```
Comment (christ66) [inline pom.xml]: Should the folders plugin be made `optional`?

**Comment by osthus-dm** [comment]
this is exactly what we need, I was considering doing this myself. any plans if and when this will be merged?

**Comment by cyrille-leclerc** [comment]
@imod do you need assistance for the last mile of this PR?

**Comment by imod** [comment]
...I get back to this soon, but I have to adapt a couple of other plugins at the same time, otherwise they will not work anymore

**Comment by osthus-dm** [comment]
awesome :+1: 
makes my day, we actually have a meeting set up for today to discuss our multiregion access strategy and this was the missing link!!!!

**Comment by imod** [comment]
@osthus-dm I just released a first beta version (2.14-beta) of this feature - it should be its available soon on the experimental-plugins-update-center: https://jenkins.io/blog/2013/09/23/experimental-plugins-update-center/

Please test it if you find some time, but also make sure to backup your configuration, because it will be migrated to the new format as soon as you have the plugin installed.

In addition to this plugin, I also have to update the managed-scripts-plugin and the ivy-plugin - because they will not work with the new version anymore. (although the ivy-plugin never worg^ked before either...)

**Comment by cyrille-leclerc** [comment]
Thanks @imod !

**Comment by imod** [comment]
just found out, there are two more plugins I have to take care of:

- https://wiki.jenkins-ci.org/display/JENKINS/Openstack+Cloud+Plugin
- https://wiki.jenkins-ci.org/display/JENKINS/Job+DSL+Plugin

**Pull request merged**